### PR TITLE
fixes for python 3

### PR DESCRIPTION
**Pull request by felmarlop**
Remember to push the tag 0.7.28.dev2 in case you merge it.

To test this you need to change the dependency link in sky/setup.py to install this library from my repo.:
```
"https://%s@github.com/felmarlop/django-social-auth/tarball/v0.7.28.dev2#egg=django-social-auth-0.7.28.dev2" % GITHUB_AUTH,

### CODE REVIEW

**Comment by felmarlop** [comment]
I pushed some 2to3 fixes , this is ready to test now.

**Comment by felmarlop** [comment]
I found another bug importing the method `urlparse` when I tried to log in with gitlab and github. I've just pushed the fix.

File: social_auth/backends/__init__.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def auth_extra_arguments(self):
         """
         backend_name = self.AUTH_BACKEND.name.upper().replace('-', '_')
         extra_arguments = setting(backend_name + '_AUTH_EXTRA_ARGUMENTS', {})
-        for key, value in extra_arguments.iteritems():
+        for key, value in list(extra_arguments.items()):
```
Comment (osroca) [inline social_auth/backends/__init__.py]: `list` is not necessary when using `.items()`, no?

```py
for key, value in a_dict.items():
    print(key, '->', value)
```

File: social_auth/backends/__init__.py [hunk 2]
```diff
@@ -483,7 +483,7 @@ def trust_root(self):
 
     def continue_pipeline(self, *args, **kwargs):
         """Continue previous halted pipeline"""
-        response = self.consumer().complete(dict(self.data.items()),
+        response = self.consumer().complete(dict(list(self.data.items())),
```
Comment (osroca) [inline social_auth/backends/__init__.py]: ditto `list` and `items()`

File: social_auth/backends/__init__.py [hunk 3]
```diff
@@ -494,7 +494,7 @@ def continue_pipeline(self, *args, **kwargs):
 
     def auth_complete(self, *args, **kwargs):
         """Complete auth process"""
-        response = self.consumer().complete(dict(self.data.items()),
+        response = self.consumer().complete(dict(list(self.data.items())),
```
Comment (osroca) [inline social_auth/backends/__init__.py]: ditto `list`

File: social_auth/backends/contrib/behance.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def extra_data(self, user, uid, response, details):
         # Pull up the embedded user attributes so they can be found as extra
         # data. See the example token response for possible attributes:
         # http://www.behance.net/dev/authentication#step-by-step
-        all_data = dict((name, value) for name, value in response.iteritems())
+        all_data = dict((name, value) for name, value in list(response.items()))
```
Comment (osroca) [inline social_auth/backends/contrib/behance.py]: ditto `list`

File: social_auth/backends/contrib/evernote.py [hunk 5]
```diff
@@ -110,7 +112,7 @@ def user_data(self, access_token, *args, **kwargs):
         """Return user data provided"""
         # drop lists
         return dict([(key, val[0]) for key, val in
-            access_token.user_info.items()])
+            list(access_token.user_info.items())])
```
Comment (osroca) [inline social_auth/backends/contrib/evernote.py]: ditto `list`

File: social_auth/backends/contrib/odnoklassniki.py [hunk 6]
```diff
@@ -98,7 +98,7 @@ def odnoklassniki_oauth_sig(data, client_secret):
     suffix = md5('{0:s}{1:s}'.format(data['access_token'],
                                      client_secret)).hexdigest()
     check_list = sorted(['{0:s}={1:s}'.format(key, value)
-                            for key, value in data.items()
+                            for key, value in list(data.items())
```
Comment (osroca) [inline social_auth/backends/contrib/odnoklassniki.py]: ditto `list`

File: social_auth/backends/contrib/odnoklassniki.py [hunk 7]
```diff
@@ -189,7 +189,7 @@ def get_response(self):
                   'apiconnection',
                   )
         response = {}
-        for fieldname in self.fields.keys():
+        for fieldname in list(self.fields.keys()):
```
Comment (osroca) [inline social_auth/backends/contrib/odnoklassniki.py:192]: `.keys()` on a dict already return a list with the name of the props in the dict

File: social_auth/backends/contrib/odnoklassniki.py [hunk 8]
```diff
@@ -204,7 +204,7 @@ def get_user_id(self, details, response):
         return response['uid']
 
     def extra_data(self, user, uid, response, details):
-        return dict([(key, value) for key, value in response.items()
+        return dict([(key, value) for key, value in list(response.items())
```
Comment (osroca) [inline social_auth/backends/contrib/odnoklassniki.py]: ditto `list`

File: social_auth/backends/facebook.py [hunk 9]
```diff
@@ -171,7 +171,7 @@ def auth_complete(self, *args, **kwargs):
     @classmethod
     def process_refresh_token_response(cls, response):
         return dict((key, val[0])
-                        for key, val in cgi.parse_qs(response).iteritems())
+                        for key, val in list(cgi.parse_qs(response).items()))
```
Comment (osroca) [inline social_auth/backends/facebook.py]: ditto `list`

**Review by osroca — APPROVED** [review]
please review those `list(adict.items())`. Even if it works, not using list must be more efficient since the whole list is not loaded in memory.

```py 
>>> d = {"one": 1, "two": 2}
>>> for key, val in d.items():
...     print(f"{key} >>> {val}")
...
one >>> 1
two >>> 2
>>> for key, val in list(d.items()):
...     print(f"{key} >>> {val}")
...
one >>> 1
two >>> 2
```

**Comment by felmarlop** [comment]
I'll push the changes in a while and replace the tag when you finish to review.

**Comment by felmarlop** [comment]
Closing... I pushed the branch https://github.com/felmarlop/django-social-auth/tree/0.7.28.dev2 and the tag v0.7.28.dev2 to bigmlcom.

**Pull request closed without merging**